bin/flutter-tizen.bat

```diff
@@ -1,22 +1,54 @@
 @ECHO off
 REM Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+REM Copyright 2014 The Flutter Authors. All rights reserved.
 REM Use of this source code is governed by a BSD-style license that can be
 REM found in the LICENSE file.
 
+REM ---------------------------------- NOTE ----------------------------------
+REM
+REM Please keep the logic in this file consistent with the logic in the
+REM `flutter-tizen` script in the same directory to ensure that Flutter continue to
+REM work across all platforms!
+REM
+REM --------------------------------------------------------------------------
+
 SETLOCAL ENABLEDELAYEDEXPANSION
 
-REM Test if python3 is installed on the host.
-where /q python3 || (
-    ECHO Error: Unable to find python3 in your PATH.
-    EXIT /B 1
+REM Detect which PowerShell executable is available on the Host
+REM PowerShell version <= 5: PowerShell.exe
+REM PowerShell version >= 6: pwsh.exe
+WHERE /Q pwsh.exe && (
+  SET powershell_exe=pwsh.exe
+) || WHERE /Q PowerShell.exe && (
+  SET powershell_exe=PowerShell.exe
+) || (
+  ECHO Error: PowerShell executable not found.                        1>&2
+  ECHO        Either pwsh.exe or PowerShell.exe must be in your PATH. 1>&2
+  EXIT 1
 )
 
-CALL python3 -V >NUL 2>NUL || (
-    ECHO Run "python3" to install python and try again.
-    EXIT /B 1
-)
+FOR %%i IN ("%~dp0..") DO SET ROOT_DIR=%%~fi
+
+SET shared_bin=%ROOT_DIR%\bin\internal\shared.bat
+
+REM Clone or update the flutter repo.
+ECHO Checking flutter repository...
```

Comment:
Redundant ECHOs?

```powershell
PS D:\Git> flutter-tizen
Checking flutter repository...
Checking flutter-tizen snapshot...
Checking flutter-tizen engine...
```

Reply:
I just added these ECHOs to find a bottle-neck because the batch script is too slow in my office environment. I will test this in other environment (at home) and remove them.